impeller/renderer/compute_command.cc

```diff
@@ -61,27 +41,20 @@ bool ComputeCommand::BindResource(
   if (!sampler || !sampler->IsValid()) {
     return false;
   }
-
-  if (!slot.HasSampler()) {
+  if (!texture || !texture->IsValid()) {
+    return false;
+  }
+  if (!slot.HasSampler() || !slot.HasTexture()) {
     return true;
   }
 
-  bindings.samplers[slot.sampler_index] = {&metadata, sampler};
-  return true;
-}
+  bindings.sampled_images[slot.sampler_index] = TextureAndSampler{
```

Comment:
I got you, this specifically is batching the hashes at creation time for the command.  When we execute we are just looping over them anyways so it doesn't matter.  SGTM.